sdk/network/Azure.ResourceManager.Network/tests/Tests/ApplicationGatewayTests.cs

```diff
@@ -508,6 +636,57 @@ private void CompareApplicationGateway(ApplicationGatewayData gw1, ApplicationGa
             Assert.AreEqual(gw1.SslPolicy.MinProtocolVersion, gw2.SslPolicy.MinProtocolVersion);
         }
 
+        private void CompareApplicationGatewayWithoutSsl(ApplicationGatewayData gw1, ApplicationGatewayData gw2)
```

Comment:
Can we share the same comparison logic with `CompareApplicationGateway` above? Maybe we can put the common part in a new method `CompareApplicationGatewayBase`.